### PR TITLE
clydegpt, translate, encryption, anti racism, impersonation, and some more stuff

### PR DESCRIPTION
**Pull request by TechFun4**
The ultimate comeback of Clyde!! uwu

### CODE REVIEW

**Review by sinjs — COMMENTED** [review]
Can you check if this still works when then plug-in gets stopped? Because you're not unsubscribing anymore

File: src/plugins/ClydeGPT/index.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            if (response.success) {
+                sendBotMessage(message.channel_id, {
+                    content: response.result,
+                });
+            } else {
+                sendBotMessage(message.channel_id, {
+                    content: "Failed to obtain AI result.",
+                });
+            }
+
```
Comment (sinjs) [inline src/plugins/ClydeGPT/index.ts]: There is an stop() handler missing. The FluxDispatcher is never unsubscribed from, which results in the plugin not being disabled correctly.

**Review by sinjs — CHANGES_REQUESTED** [review]
Fix linting and formatting using `pnpm run lint:fix`

**Comment by TechFun4** [comment]
fixed and upgraded clyde

File: src/plugins/ClydeGPT/index.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        
+    if (!message.content.includes("@Clyde")) {
+        return;
+    }
+
+    if (message.author.id == UserStore.getCurrentUser().id && !msg?.sendMessageOptions?.nonce) {
+        return;
+    }
+
+    if (message.author?.bot) {
```
Comment (Airbus-A330) [inline src/plugins/ClydeGPT/index.ts]: Is this necessary? We check for author ID a few lines up.

**Review by Airbus-A330 — CHANGES_REQUESTED** [review]
Questions about message author.

File: src/plugins/ClydeGPT/index.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        type: "MESSAGE_CREATE", 
+        channelId: message.channel_id,
+        message: {
+            id: (BigInt(Date.now() - 1420070400000) << 22n).toString(),
+            channel_id: message.channel_id,
+            author: {
+                id: "1081004946872352958"
+            },
+            content: response.result,
+            timestamp: new Date().toISOString()
```
Comment (Airbus-A330) [inline src/plugins/ClydeGPT/index.ts]: Remove `.toISOString()`, `new Date()` is sufficient.

**Review by Airbus-A330 — CHANGES_REQUESTED** [review]
Fix minor things.

File: src/plugins/ClydeGPT/index.ts (hunk 2 above)
Comment (TechFun4) [reply src/plugins/ClydeGPT/index.ts]: yes, Clyde should not respond to bots

**Comment by Airbus-A330** [comment]
I've made some changes about how the reply is triggered. The above issues in my last review have not been completed yet.

File: src/plugins/ClydeGPT/index.ts (hunk 3 above)
Comment (sinjs) [reply src/plugins/ClydeGPT/index.ts]: Isn't this an API type? it must be a string right

**Comment by sinjs** [comment]
Tell me when you're ready to merge :)

File: src/plugins/ClydeGPT/index.ts (hunk 2 above)
Comment (sinjs) [reply src/plugins/ClydeGPT/index.ts]: But you're already checking if the message author equals to the current user? If I understand the code correctly

**Comment by sinjs** [comment]
please don't force push my changes away again 💔

**Comment by sinjs** [comment]
Fix the formatting, `pnpm lint:fix` and then manually fix the remaining issues if any

**Comment by sinjs** [comment]
what even is this anymore

**Comment by Airbus-A330** [comment]
I honestly have no idea, bro doesn't know how to make branches to separate features...

**Comment by sinjs** [comment]
u cant u just gotta make a new pr but since this isnt merged yet its weird

**Pull request merged**